src/java.desktop/macosx/classes/sun/font/CCharToGlyphMapper.java

```diff
@@ -88,6 +88,31 @@ public synchronized boolean charsToGlyphsNS(int count,
         return false;
     }
 
+    private static int unicodeToUnits(int unicode, int dstOffset, char[] dst) {
+        if (unicode >= 0x10000) {
```

Comment:
Why can't you use public API that already does this ? [ ](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Character.html#toChars(int,char%5B%5D,int)) ?

Reply:
Which API exactly? I didn't know there is any, so I added this as a helper method.

Reply:
I suppose it's https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Character.html#charCount(int)

Reply:
Fixed